src/api/tojs.ts

```diff
@@ -58,4 +58,20 @@ export function toJS(source, detectCycles: boolean = true, __alreadySeen: [any,
 export function toJSON(source, detectCycles: boolean = true, __alreadySeen: [any, any][] = null) {
 	deprecated("toJSON is deprecated. Use toJS instead");
 	return toJS.apply(null, arguments);
-}
\ No newline at end of file
+}
+
+function isDOMNode(source) {
+	const isDOMAvailable = (
+		typeof document === "object" &&
+		document &&
+		document.createElement &&
+		typeof HTMLElement === "function" &&
+		typeof Node === "function"
+	);
```

Comment:
All those checks are unnecessarily executed each time `isDOMNode` is called. In my variant (#566) they are called only once. Not a bid deal, but what for?